apps/src/code-studio/components/progress/LessonLockDialog.jsx

```diff
@@ -11,264 +10,287 @@ import SectionSelector from './SectionSelector';
-      return;
+  useEffect(() => {
+    if (!saving) {
+      setlockStatuses(initialLockStatus);
     }
-    this.setState({
-      lockStatus: nextProps.initialLockStatus
-    });
-  }
+  }, [initialLockStatus]);
 
-  setAllLockStatus(lockStatus) {
-    this.setState({
-      lockStatus: this.state.lockStatus.map(item =>
-        Object.assign({}, item, {
-          lockStatus
-        })
-      )
-    });
-  }
+  //
+  // Event handlers
+  //
+  const setAllLockStatus = lockStatus => {
+    setlockStatuses(currentLockStatuses =>
+      currentLockStatuses.map(item => ({...item, lockStatus}))
+    );
+  };
 
-  allowEditing = () => this.setAllLockStatus(LockStatus.Editable);
+  const allowEditing = () => setAllLockStatus(LockStatus.Editable);
 
-  lockLesson = () => this.setAllLockStatus(LockStatus.Locked);
+  const lockLesson = () => setAllLockStatus(LockStatus.Locked);
 
-  showAnswers = () => this.setAllLockStatus(LockStatus.ReadonlyAnswers);
+  const showAnswers = () => setAllLockStatus(LockStatus.ReadonlyAnswers);
 
-  viewSection = () => {
+  const viewSection = () => {
     const assessmentsUrl = teacherDashboardUrl(
-      this.props.selectedSectionId,
+      selectedSectionId,
       '/assessments'
     );
     window.open(assessmentsUrl, '_blank', 'noopener,noreferrer');
   };
 
-  handleRadioChange = event => {
+  const handleRadioChange = event => {
     const modifiedIndex = parseInt(event.target.name, 10);
     const value = event.target.value;
 
-    // Do this in a setTimeout so that the event has time to finish processing
-    // (i.e. so that checkbox is updated) before we call setState and rerender
-    setTimeout(() => {
-      this.setState({
-        lockStatus: this.state.lockStatus.map((item, index) => {
-          if (index !== modifiedIndex) {
-            return item;
-          }
-          return Object.assign({}, item, {
-            lockStatus: value
-          });
-        })
-      });
-    }, 0);
+    setlockStatuses(currentLockStatuses =>
+      currentLockStatuses.map((item, index) => {
+        if (index !== modifiedIndex) {
+          return item;
+        }
+        return {...item, lockStatus: value};
+      })
+    );
   };
 
-  handleSave = () => {
-    this.props.saveDialog(this.props.selectedSectionId, this.state.lockStatus);
+  const handleSave = () => {
+    saveDialog(selectedSectionId, lockStatuses);
   };
 
-  render() {
-    const responsiveHeight = {
-      maxHeight: window.innerHeight * 0.8 - 100
-    };
-    const hasSelectedSection = this.props.selectedSectionId !== NO_SECTION;
-    const hiddenUnlessSelectedSection = hasSelectedSection ? {} : styles.hidden;
-    return (
-      <BaseDialog
-        isOpen={this.props.isOpen}
-        handleClose={this.props.handleClose}
+  //
+  // Rendering helpers that each render a section of the dialog
+  //
+  const hasSelectedSection = selectedSectionId !== NO_SECTION;
+  const hiddenUnlessSelectedSection = hasSelectedSection ? {} : styles.hidden;
```

Comment:
I know this was already existing, but does there seem to be a reason that we're hiding elements through CSS instead of just not rendering them?